xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-api/src/test/java/org/xwiki/security/authorization/cache/internal/DefaultSecurityCacheTest.java

```diff
@@ -813,20 +815,26 @@ void testCacheEvictedEntries() throws Exception
         entries.putAll(InsertEntities());
         entries.putAll(InsertAccess());
 
-        final Map<SecurityReference, String> keys = new HashMap<SecurityReference, String>();
-
         for (Map.Entry<String, SecurityEntry> entry : entries.entrySet()) {
-            if (entry.getValue() instanceof SecurityRuleEntry) {
-                keys.put(entry.getValue().getReference(), entry.getKey());
+            // Remove all "internal" entries and remove one access entry for the docRef/userRef.
+            if (!(entry.getValue() instanceof SecurityAccessEntry)
+                || (entry.getValue().getReference() == this.docRef
+                && ((SecurityAccessEntry) entry.getValue()).getUserReference() == this.userRef))
+            {
+                this.cache.remove(entry.getKey());
             }
         }
 
-        removerTest(entries, new Remover()
+        // For a GC to try emptying the reference map as much as possible to purge the docRef/userRef access entry.
+        // All "internal" entries should be kept as they should be referenced through parents of the access entries
+        // still in the cache.
+        System.gc();
```

Comment:
```suggestion
        // Force a GC to try emptying the reference map as much as possible to purge the docRef/userRef access entry.
        // All "internal" entries should be kept as they should be referenced through parents of the access entries
        // still in the cache.
        System.gc();
```

Also, possibly because I don't know well this part of the code but I don't get from the comments why calling the gc is relevant. Asking as it is quite unusual to explicitly call it.

Reply:
> I don't get from the comments why calling the gc is relevant

Well it's explained in the comment, it's to trigger the emptying of the `ReferenceMap`, I didn't know about such map implementation, but it's a map that aims at getting its elements removed thanks to the GC, see: https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/ReferenceMap.html

Reply:
Yeah, so basically this map stores only weak references to the values (as I configured it that way) so values will be garbage collected if there are no other strong references to them. The garbage collector notifies the map which values have been garbage collected so the map can properly remove them.

Reply:
> Well it's explained in the comment, it's to trigger the emptying of the `ReferenceMap`

ok, but why do you need to explicitly empty it at this exact point in time.
Also, garbage collection the whole VM just to clean a single collection feels super heavy.

Repeating: I don't know the domain so I might be asking silly questions :)

Reply:
Sorry, I'd missed it was for a test... Perfectly fine here of course :)